File: Languages/Nim/bee_movie.nim
# Hello, World!
echo "Hello, World!"

import os, strutils, algorithm, tables

type
  LetterCount = ref object
    letter: char
    count: int

proc analyzeBeeMovie() =
  let scriptPath = "../beemoviescript.txt"
  
  if not fileExists(scriptPath):
    echo "File not found: " & scriptPath
    return
  
  echo "Bee Movie Script:"
  echo "--------------------------------------------------"
  
  var letterCounts = initCountTable[char]()
  var totalLetters = 0
  
  # Read file line by line and print each line
  for line in scriptPath.lines():
    echo line
    
    # Count letters
    for char in line:
      if char.isAlphaAscii():
        let lower = char.toLowerAscii()
        letterCounts.inc(lower)
        totalLetters += 1
  
  echo "--------------------------------------------------"
  echo "Analysis complete."
  
  if totalLetters > 0:
    # Convert to array and sort
    var counts: seq[LetterCount]
    for letter, count in letterCounts.pairs():
      counts.add(LetterCount(letter: letter, count: count))
    
    counts.sort(proc(a, b: LetterCount): int = cmp(b.count, a.count))
    
    echo "\nTop 3 most commonly used letters:"
    for i in 0..<min(3, counts.len):
      let item = counts[i]
      echo $(i+1) & ". '" & item.letter & "': " & $item.count & " times"
  else:
    echo "No letters found in the script."

analyzeBeeMovie()

File: Languages/Lua/bee_movie.lua
-- Hello, World!
print("Hello, World!")

function analyze_bee_movie()
    local script_path = "../beemoviescript.txt"
    
    local file = io.open(script_path, "r")
    if not file then
        print("File not found: " .. script_path)
        return
    end
    
    print("Bee Movie Script:")
    print("--------------------------------------------------")
    
    local letter_counts = {}
    local total_letters = 0
    
    -- Read file line by line and print each line
    for line in file:lines() do
        print(line)
        
        -- Count letters
        for char in line:gmatch(".") do
            if char:match("[a-zA-Z]") then
                local lower = char:lower()
                letter_counts[lower] = (letter_counts[lower] or 0) + 1
                total_letters = total_letters + 1
            end
        end
    end
    
    file:close()
    
    print("--------------------------------------------------")
    print("Analysis complete.")
    
    if total_letters > 0 then
        -- Convert to array and sort
        local sorted = {}
        for letter, count in pairs(letter_counts) do
            table.insert(sorted, {letter = letter, count = count})
        end
        
        table.sort(sorted, function(a, b) return a.count > b.count end)
        
        print("\nTop 3 most commonly used letters:")
        for i = 1, math.min(3, #sorted) do
            local item = sorted[i]
            print(string.format("%d. '%s': %d times", i, item.letter, item.count))
        end
    else
        print("No letters found in the script.")
    end
end

analyze_bee_movie()
